dbusutil/dbustest/dbustest.go

```diff
@@ -211,23 +222,30 @@ func (a *testAuth) HandleData(data []byte) (resp []byte, status dbus.AuthStatus)
 	return []byte(""), dbus.AuthOk
 }
 
-// Connection returns a DBus connection for writing unit tests.
-//
-// The handler function is called for each message sent to the bus. It can
-// return any number of messages to send in response. The counter aids in
-// testing a sequence of messages that is expected.
-func Connection(handler DBusHandlerFunc) (*dbus.Conn, error) {
-	conn, err := dbus.NewConn(newTestDBusStream(handler))
+type InjectMessageFunc func(msg *dbus.Message)
+
+func InjectableConnection(handler DBusHandlerFunc) (*dbus.Conn, InjectMessageFunc, error) {
```

Comment:
this could use a doc-comment, especially an example of what situation one would want to use InjectableConnection() for instead of Connection()

Reply:
added